core-bundle/src/Pagination/PaginationFactory.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Contao\CoreBundle\Pagination;
+
+use Symfony\Component\HttpFoundation\RequestStack;
+
+class PaginationFactory implements PaginationFactoryInterface
+{
+    public function __construct(
+        private readonly RequestStack $requestStack,
+        private readonly int $defaultRange = 7,
+    ) {
+    }
+
+    public function create(string $param, int $total, int $perPage, int|null $pageRange = null, bool $throw = true): PaginationInterface
```

Comment:
```suggestion
    public function create(PaginationConfig $config): PaginationInterface
```

```php
class PaginationConfig
{
    // mandatory parameters in the constructor
    public function __construct(private string $queryParameterName, private int $total, private int $perPage)
    {}

    // optional settings as withers
    public function withPageRange(...) {}
    public function withThrow(...) {}

    // If you want, add factories for use cases you need often:
    public static createFromStuffINeedOften(...): self {}
}
```

I would prefer to have this object in the interface. It will allow us to add things very easily without breaking the interface in the future. What if e.g. you want to be able to accept a `Request` instead of fetching it from the `RequestStack`? The interface becomes really ugly if we add 5 arguments more. With an object, this is super easy.